src/common.rkt

```diff
@@ -163,12 +163,44 @@
    [(< (max ms min-unit) 3600000) (format "~amin" (/ (round (/ ms 6000.0)) 10))]
    [else (format "~ahr" (/ (round (/ ms 360000.0)) 10))]))
 
-(define (format-bits r #:sign [sign #f] #:unit [unit? #f])
-  (define unit (if unit? "b" ""))
-  (cond
-   [(not r) ""]
-   [(and (> r 0) sign) (format "+~a~a" (/ (round (* r 10)) 10) unit)]
-   [else (format "~a~a" (/ (round (* r 10)) 10) unit)]))
+(define (format-bits r repr #:sign [sign #f] #:unit [unit #f])
+  (cond 
+    [(zero? (representation-total-bits repr)) (error 'failed)]
+    [(not r) ""]
+    [else
+      (define unit- (if unit unit ""))
+      (define percent
+        (~r 
+        (cond
+          [unit (* (/ r (representation-total-bits repr)) 100)]
+          [else (/ (round (* r 10)) 10)]) 
+        #:precision 2))
+
+      (cond
+      [(and (> r 0) sign) (format "+~a~a" percent unit-)]
+      [else (format "~a~a" percent unit-)])]))
+
+(define (format-error r repr #:sign [sign #f] #:unit [unit #f])
+  (cond 
+    [(zero? (representation-total-bits repr)) (error 'failed)]
+    [(not r) ""]
+    [else
+      (define unit- (if unit unit ""))
+      (define percent (~r (* (/ r (representation-total-bits repr)) 100) #:precision 2))
+
+      (cond
+      [(and (> r 0) sign) (format "+~a~a" percent unit-)]
+      [else (format "~a~a" percent unit-)])]))
+
+(define (format-cost r repr #:sign [sign #f])  
+  (cond 
+    [(zero? (representation-total-bits repr)) (error 'failed)]
+    [(not r) ""]
+    [else
+      (define val (~r (/ (round (* r 10)) 10) #:precision 2))
+      (cond
+      [(and (> r 0) sign) (format "+~a" val)]
+      [else (format "~a" val)])]))
```

Comment:
Check for `zero?` should be removed.

Reply:
In the meeting I think Zack said it was ok if we threw an error if it happened.

Reply:
If we are going to throw an assert we need some library and I don't know which one it is.

Reply:
I think it should be removed. Under no circumstance should it be zero.

Reply:
Ok. Will do.